async_service/asyncio.py

```diff
@@ -246,10 +247,11 @@ def run_task(
         current_task = asyncio.Task.current_task()
         if not _root and current_task not in self._service_task_dag:
             raise Exception(f"TODO: unknown task {current_task}")
+        if name is None:
+            name = repr(async_fn)
```

Comment:
How would you feel about using `async_fn.__name__` here instead? For someone who's been looking at those logs a log recently, I find the fully qualified names make them a lot harder to read

Reply:
:+1: I'll do an incremental thing that still falls back to `repr(...)` because I recall there being some cases where `__name__` isn't available.